modules/affile/file-opener.c

```diff
@@ -182,6 +182,86 @@ file_opener_open_fd(FileOpener *self, const gchar *name, FileDirection dir, gint
   return FILE_OPENER_RESULT_SUCCESS;
 }
 
+void
+file_opener_symlink(FileOpener *self, const gchar *name, const gchar *target)
+{
+  cap_t saved_caps;
+  gboolean do_symlink = FALSE;
+
+  msg_trace("file_opener_symlink",
+            evt_tag_str("filename", name),
+            evt_tag_str("target", target));
+
+  /* do not proceed if name already exists and is not a symlink; also, do not
+   * disturb the symlink unnecessarily (avoids annoying a running "tail -f",
+   * etc.) */
+
+  size_t n = strlen(target);
+  if (n > SSIZE_MAX - 1) return; /* avoid overflowing ssize_t */
+  gchar *s = malloc(n + 1);
+  if (s == NULL) return;
+  ssize_t i = readlink(name, s, n + 1);
```

Comment:
If we are using the readlink and malloc function, there can be a problem. The readlink function can be silently truncate the buff and that's can make some inconsistent.
Official manually of readlink: `... It will (silently) truncate the contents  (to a  length of bufsiz characters), in case the buffer is too small to hold all of the contents. ....`
So If you are using the g_file_read_link function instead of readlink & malloc, that will solve the problem.